Client/Frontend/Browser/Tabs/TabGroupsManager.swift

```diff
@@ -6,137 +6,287 @@ import Foundation
 import Shared
 import MozillaAppServices
 
+struct ASGroup<T> {
+    let searchTerm: String
+    let groupedItems: [T]
+    let timestamp: Timestamp
+}
+
 class TabGroupsManager {
 
-    // Create URL groups from metadata
+    // Leaving this for reference for when we need to add URL to group functionality.
```

Comment:
@electricRGB How would you feel about removing these as its already part of main branch and if we need to look at this then we can come back to this PR?

It'll be good for cleanup

Reply:
sure. No worries.